Binärdateien osdteletext-0.4.1/display.o and /home/michael/Desktop/osdteletext-0.4.1/display.o sind verschieden.
diff -urN osdteletext-0.4.1/i18n.c /home/michael/Desktop/osdteletext-0.4.1/i18n.c
--- osdteletext-0.4.1/i18n.c	2004-06-17 22:01:40.000000000 +0200
+++ /home/michael/Desktop/osdteletext-0.4.1/i18n.c	2005-02-07 11:00:47.000000000 +0100
@@ -46,7 +46,7 @@
     "ÝÕ ÝÐÙÔÕÝÐ",
   },
   { "Teletext (OSD)",
-    "Teletext (OSD)",
+    "Videotext (OSD)",
     "",// TODO
     "Teletext (OSD)",// TODO
     "",// TODO
Binärdateien osdteletext-0.4.1/i18n.o and /home/michael/Desktop/osdteletext-0.4.1/i18n.o sind verschieden.
diff -urN osdteletext-0.4.1/Makefile /home/michael/Desktop/osdteletext-0.4.1/Makefile
--- osdteletext-0.4.1/Makefile	2004-09-21 18:08:17.000000000 +0200
+++ /home/michael/Desktop/osdteletext-0.4.1/Makefile	2005-02-07 11:00:47.000000000 +0100
@@ -15,8 +15,6 @@
 
 ### The C++ compiler and options:
 
-CXX      ?= g++
-CXXFLAGS ?= -O2 -g -Wall -Woverloaded-virtual
 # for debugging:
 #CXXFLAGS ?= -O0 -g -Wall -Woverloaded-virtual
 
Binärdateien osdteletext-0.4.1/menu.o and /home/michael/Desktop/osdteletext-0.4.1/menu.o sind verschieden.
Binärdateien osdteletext-0.4.1/osdteletext.o and /home/michael/Desktop/osdteletext-0.4.1/osdteletext.o sind verschieden.
diff -urN osdteletext-0.4.1/txtbitmap.c /home/michael/Desktop/osdteletext-0.4.1/txtbitmap.c
--- osdteletext-0.4.1/txtbitmap.c	2004-07-06 22:41:38.000000000 +0200
+++ /home/michael/Desktop/osdteletext-0.4.1/txtbitmap.c	2005-02-07 11:00:47.000000000 +0100
@@ -189,8 +189,9 @@
    
    char zeichen[9];
    time_t t=time(0);
-   struct tm* loct=localtime(&t);
-   sprintf(zeichen, "%02d:%02d:%02d", loct->tm_hour, loct->tm_min, loct->tm_sec);
+   struct tm loct;
+   localtime_r(&t, &loct);
+   sprintf(zeichen, "%02d:%02d:%02d", loct.tm_hour, loct.tm_min, loct.tm_sec);
    
    
    // paste them into the bitmap
Binärdateien osdteletext-0.4.1/txtbitmap.o and /home/michael/Desktop/osdteletext-0.4.1/txtbitmap.o sind verschieden.
Binärdateien osdteletext-0.4.1/txtfont.o and /home/michael/Desktop/osdteletext-0.4.1/txtfont.o sind verschieden.
diff -urN osdteletext-0.4.1/txtrecv.c /home/michael/Desktop/osdteletext-0.4.1/txtrecv.c
--- osdteletext-0.4.1/txtrecv.c	2004-07-24 22:35:09.000000000 +0200
+++ /home/michael/Desktop/osdteletext-0.4.1/txtrecv.c	2005-02-07 11:00:47.000000000 +0100
@@ -9,6 +9,8 @@
  ***************************************************************************/
 
 
+#include <dirent.h>
+
 #include "txtrecv.h"
 #include "tables.h"
 #include "setup.h"
@@ -19,7 +21,6 @@
 #include <pthread.h> 
 #include <signal.h> 
 #include <errno.h>
-#include <dirent.h>
 #include <sys/vfs.h> 
 #include <sys/types.h>
 #include <sys/stat.h>
@@ -39,10 +40,10 @@
    DIR *top=opendir(root);
    int pagesDeleted=0;
    if (top) {
-      struct dirent *chandir;
+      struct dirent *chandir, path;
       struct stat chandirstat;
       char fullPath[PATH_MAX];
-      while ( (chandir=readdir(top)) ) {
+      while ( (!readdir_r(top, &path, &chandir) && chandir != NULL) ) {
          if (strcmp(chandir->d_name, "..")==0 || strcmp(chandir->d_name, ".")==0)
             continue;
          snprintf(fullPath, PATH_MAX, "%s/%s", root, chandir->d_name);
@@ -65,10 +66,10 @@
    bool hadError=false;
    int bytesDeleted=0, filesize;
    if (d) {
-      struct dirent *txtfile;
+      struct dirent *txtfile, path;
       struct stat txtfilestat;
       char fullPath[PATH_MAX];
-      while ( (txtfile=readdir(d)) ) {
+      while ( (!readdir_r(d, &path, &txtfile) && txtfile != NULL) ) {
          int len=strlen(txtfile->d_name);
          //check that the file end with .vtx to avoid accidents and disasters
          if (strcmp(txtfile->d_name+len-4, ".vtx")==0) {
@@ -150,9 +151,9 @@
    int haveDir=0;
    DIR *top=opendir(getRootDir());
    if (top) {
-      struct dirent *chandir;
+      struct dirent *chandir, path;
       struct stat chandirstat;
-      while ( (chandir=readdir(top)) ) {
+      while ( (!readdir_r(top, &path, &chandir) && chandir != NULL) ) {
          if (strcmp(chandir->d_name, "..")==0 || strcmp(chandir->d_name, ".")==0)
             continue;
          snprintf(fullPath, PATH_MAX, "%s/%s", getRootDir(), chandir->d_name);
@@ -182,12 +183,12 @@
 }
 
 void Storage::getFilename(char *buffer, int bufLength, PageID page) {
-   snprintf(buffer, bufLength, "%s/%s/%03x_%02x.vtx", getRootDir(), page.channel.ToString(), page.page, page.subPage);
+   snprintf(buffer, bufLength, "%s/%s/%03x_%02x.vtx", getRootDir(), (const char*)page.channel.ToString(), page.page, page.subPage);
 }
 
 void Storage::prepareDirectory(tChannelID chan) {
    free(currentDir);
-   asprintf(&currentDir, "%s/%s", root, chan.ToString());
+   asprintf(&currentDir, "%s/%s", root, (const char*)chan.ToString());
    MakeDirs(currentDir, 1);
    failedFreeSpace=false;
 }
@@ -387,7 +388,7 @@
 void PackedStorage::getFilename(char *buffer, int bufLength, PageID page) {
    //This is a different scheme: page 576_07 will have the name 570s.vtx, the same as e.g. 571_01 or 575_00
    //Think of "the five hundred seventies"
-   snprintf(buffer, bufLength, "%s/%s/%03xs.vtx", getRootDir(), page.channel.ToString(), (page.page & 0xFF0));
+   snprintf(buffer, bufLength, "%s/%s/%03xs.vtx", getRootDir(), (const char*)page.channel.ToString(), (page.page & 0xFF0));
 }
 
 StorageHandle PackedStorage::openForWriting(PageID page) {
@@ -665,7 +666,7 @@
 */
 
 cTxtReceiver::cTxtReceiver(int TPid, tChannelID chan)
- : cReceiver(0, -1, 1, TPid), 
+ : cReceiver(0, -1, TPid, NULL, NULL, NULL), 
    chan(chan), TxtPage(0), buffer((188+60)*75), running(false)
 {
    Storage::instance()->prepareDirectory(chan);
